src/sdk/namespace_s3.js

```diff
@@ -119,16 +119,30 @@ class NamespaceS3 {
     // OBJECT READ //
     /////////////////
 
+    _set_md_conditions(params, request) {
+        if (params.md_conditions) {
+            request.IfMatch = params.md_conditions.if_match_etag;
+            request.IfNoneMatch = params.md_conditions.if_none_match_etag;
+            if (params.md_conditions.if_modified_since) {
+                request.IfModifiedSince = new Date(params.md_conditions.if_modified_since);
+            }
+            if (params.md_conditions.if_unmodified_since) {
+                request.IfUnmodifiedSince = new Date(params.md_conditions.if_unmodified_since);
+            }
+        }
+    }
+
     async read_object_md(params, object_sdk) {
         try {
             dbg.log0('NamespaceS3.read_object_md:', this.bucket, inspect(params));
-            const request = { Key: params.key, Range: `bytes=0-${config.INLINE_MAX_SIZE - 1}` };
+            const request = { Key: params.key, Range: `bytes=0-${config.INLINE_MAX_SIZE - 1}`, PartNumber: params.part_number };
```

Comment:
Do you think that part_number makes sense here in the inline read?
Seems like this will either be ignored with Range or have some other unexpected behavior...